### DISCUSSION TITLE
Upgrade from v2.9.3 to 2.12.x

### DISCUSSION THREAD

**Discussion by mesutcelik** [argoproj/argo-workflows#5548] [Q&A] [answered]
Hi,

We are using Argo v2.9.3 and want to upgrade to the latest 2.12.x. Is there any upgrade documentation covering how to do it?

**Comment by terrytangyuan** [comment, marked as answer]
I don’t think there is any documentation on this but upgrading from v2.9 to v2.12 should be pretty easy compared to v3.0 or from v2.4 to v2.12.

Note that there are migration notes from v2 on v3.0 release notes [here](https://github.com/argoproj/argo-workflows/releases/tag/v3.0.0) if you’d like to try out v3.0. There are lots of great new features in v3.0!

**Reply by mesutcelik** [reply]
As Argo is using a database, I thought that there is a migration guide including DB part. If 2.12 is compatible with 2.9 version DB schema then I guess it is as easy as updating argo docker images.

These are my current versions:
argo-server: `argoproj/argocli:v2.9.3` 
workflow-controller: `argoproj/workflow-controller:v2.9.3`

**Reply by terrytangyuan** [reply]
I guess this is a lack of docs in general (I also mentioned DB migration earlier in https://github.com/argoproj/argo-workflows/issues/5329#issuecomment-797782256).

**Reply by mesutcelik** [reply]
Ok so no DB migration is needed for the time being to move from v2.9.3 to 2.12.1 or v3.0.2 ?

**Reply by terrytangyuan** [reply]
For v2.9.3 to 2.12.1, I think so but uncertain about v3.0.2. Right now the best way to find out is to test the migration in your testing environment.

**Reply by mesutcelik** [reply]
Thanks @terrytangyuan , It is much appreciated!
